src/validation_workflow/tar/validation_tar.py

```diff
@@ -28,15 +28,13 @@ def __init__(self, args: ValidationArgs) -> None:
         self.osd_process = Process()
 
     def download_artifacts(self) -> bool:
-        architectures = ["x64", "arm64"]
         for project in self.args.projects:
-            for architecture in architectures:
-                url = f"{self.base_url}{project}/{self.args.version}/{project}-{self.args.version}-linux-{architecture}.tar.gz"
-                if ValidateTar.is_url_valid(url) and ValidateTar.download(url, self.tmp_dir):
-                    logging.info(f"Valid URL - {url} and Download Successful !")
-                else:
-                    logging.info(f"Invalid URL - {url}")
-                    raise Exception(f"Invalid url - {url}")
+            url = f"{self.base_url}{project}/{self.args.version}/{project}-{self.args.version}-linux-{self.args.arch}.tar.gz"
```

Comment:
Where is the self.args.arch coming from, could you point me to the code?

Reply:
That's from the user input initialized in the init function could be either x64 or arm64 on which the workflow runs
https://github.com/opensearch-project/opensearch-build/blob/6191810fc47f8aa903a99ac6688da065d01bef70/src/validation_workflow/tar/validation_tar.py#LL23C16-L23C16